systemtest/produce_consume_group_test.go

```diff
@@ -0,0 +1,327 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package systemtest
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/elastic/apm-data/model"
+	apmqueue "github.com/elastic/apm-queue"
+	"github.com/elastic/apm-queue/codec/json"
+	"github.com/elastic/apm-queue/kafka"
+	"github.com/elastic/apm-queue/pubsublite"
+)
+
+// TestProduceConsumeDelivery verifies that a failure to process an event won't affect the
+// other events in the same batch.
+func TestProduceConsumeDelivery(t *testing.T) {
+	testCases := map[string]struct {
+		deliveryType         apmqueue.DeliveryType
+		events               int
+		replay               int
+		expectedRecordsCount int
+		timeout              time.Duration
+	}{
+		"at most once": {
+			deliveryType:         apmqueue.AtMostOnceDeliveryType,
+			events:               100,
+			replay:               1,
+			expectedRecordsCount: 199,
+			timeout:              60 * time.Second,
+		},
+		"at least once": {
+			deliveryType:         apmqueue.AtLeastOnceDeliveryType,
+			events:               100,
+			replay:               1,
+			expectedRecordsCount: 199,
+			timeout:              60 * time.Second,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run("Kafka/"+name, func(t *testing.T) {
+			logger := NoLevelLogger(t, zap.ErrorLevel)
```

Comment:
This level logger can't be used since the consumer will return an error. All these tests are failing

Reply:
Should be fixed